bioblend/_tests/GalaxyTestBase.py

```diff
@@ -25,3 +26,23 @@ def _test_dataset(self, history_id, contents="1\t2\t3", **kwds):
     def _wait_and_verify_dataset(self, dataset_id, expected_contents, timeout_seconds=BIOBLEND_TEST_JOB_TIMEOUT):
         dataset_contents = self.gi.datasets.download_dataset(dataset_id, maxwait=timeout_seconds)
         self.assertEqual(dataset_contents, expected_contents)
+
+    def _wait_invocation(self, invocation, wait_steps=False):
```

Comment:
Should we maybe move and expose this as a convenience `wait_for_invocation()` method of `InvocationClient` ?

Reply:
Instead of asserting that it was scheduled, would it then be better to return `True` if the scheduling was successful, and `False` if time ran out?

Reply:
In other parts of BioBlend we raise a specific exception (e.g. `DatasetTimeoutException`), like [requests](https://requests.readthedocs.io/) does.

Reply:
> Should we maybe move and expose this as a convenience `wait_for_invocation()` method of `InvocationClient` ?

Sounds good to me

Reply:
It has been moved to InvocationClient